### PR TITLE
Fixes sbt#121: Suite task naming no longer crashes; or mangles vintage test names

### PR DESCRIPTION
**Pull request by dji**
Fixes (and tests!) the two related issues in #121:
- In some situations `@Suite`s were used, a RuntimeException would be thrown when trying to report the test status: `Test class path.to.my.TestClass is not enclosed by path.to.my.SuiteClass`
- Junit 4 tests run as part of a suite had their names mangled

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -13,6 +13,7 @@ object Dependencies {
   val testInterface = "org.scala-sbt" % "test-interface" % testInterfaceVer
   val junitJupiterParams = "org.junit.jupiter" % "junit-jupiter-params" % junitJupiterVer
   val junitVintageEngine = "org.junit.vintage" % "junit-vintage-engine" % junitVintageVer
+  val junitPlatformSuite = "org.junit.platform" % "junit-platform-suite" % junitPlatformVer
```
Comment (dji) [inline project/Dependencies.scala:16]: Alternatively, only `junit-platform-suite-api` and `junit-platform-suite-engine` are actually needed, if you prefer more specific dependencies

**Review by eed3si9n — APPROVED** [review]
Thanks!

**Comment by dji** [comment]
Fixed the formatting issue and merged in the main branch - what are the next steps to get this merged in?

**Comment by eed3si9n** [comment]
Rerunning CI now.

**Comment by dji** [comment]
And you did a release too! :)  Thanks.

**Pull request merged**